test-lib.sh

```diff
@@ -109,15 +84,34 @@ ct_build_image_and_parse_id() {
   local ret_val
   local dockerfile
   local command
+  local pid_build
+  local pid_sleep
+  local sleep_time
   log_file="$(mktemp)"
-  ret_val=1
+  sleep_time="10m"
   [ -n "$1" ] && dockerfile="-f $1"
   command="$(echo "docker build --no-cache $dockerfile $2" | tr -d "'")"
-
-  if $command > "$log_file" 2>&1; then
+  # running command in subshell, the subshell in background, storing pid to variable
```

Comment:
Yes, I already explained it here:
https://github.com/sclorg/container-common-scripts/pull/338#discussion_r1142117196

I can explain further if this explanation is not sufficient.

Reply:
In case of some container build, like `s2i-ruby-container` takes a long time, then we should fix the corresponding container and set up a hard timeout. What about if time out exceeds more than 10m e.g. in the case of s2i-python-container, we will increase the timeout again? The rest of this pull request is fine, but this part of code brings me a lot of issues in the future. What about to add also into test suite s2i-ruby-container? I would not complicate container build so much. Let's build it.

Reply:
https://github.com/sclorg/s2i-ruby-container/issues/469 This is the issue causing never ending build.

Reply:
We should really have some timeout, even though 10m is rather arbitrary and I expect we will have to increase it at some point after we hit longer builds for some reason.
Without the timeout we are just hogging testing farm resources without good reason if we hit such an issue at buildtime.

Reply:
As this is the last opened issue I see on this PR, can I now merge it?

Reply:
Let's get merge it.